extensions/src/org/pentaho/platform/plugin/services/security/userrole/ldap/DefaultLdapUserRoleListService.java

```diff
@@ -155,6 +165,13 @@ public void afterPropertiesSet() throws Exception {
     for ( GrantedAuthority role : results ) {
       roles.add( role.getAuthority() );
     }
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
```

Comment:
The extra role are added to the user's role that exist on the third-party security provider. So those role are being added to the user. Are you still concerned here ?